### PR TITLE
add documentation to char.zp

### PR DESCRIPTION
**Pull request by 4www**
Is it the way you would like it?
can I improve some aspects?

### CODE REVIEW

**Comment by hellerve** [comment]
This looks pretty neat already, I attached some comments.

File: char.zp [hunk 1]
```diff
@@ -1,59 +1,179 @@
-(define (char:cmp? cmp a b) "compares two chars with a compare option cmp"
+(define (char:cmp? cmp a b)
+  "compares two chars with a compare option cmp.
```
Comment (hellerve) [inline char.zp:2]: Please add an empty line between the summary and the meta information.

**Review by hellerve — CHANGES_REQUESTED** [review]

File: char.zp [hunk 2]
```diff
@@ -1,59 +1,179 @@
...
+  params:
+   - compare option
```
Comment (hellerve) [inline char.zp]: Parameters should adhere to this format:
`- param_name: description (including type and whether it is optional etc.)`

**Review by hellerve — CHANGES_REQUESTED** [review]

File: char.zp (hunk 1 above)
Comment (4www) [reply char.zp:2]: sure. What's that for?

File: char.zp (hunk 2 above)
Comment (hugovieilledent) [inline char.zp]: ok

Comment (4www) [reply char.zp]: for exemple?
- compare_option: a compare function [string](required)
  What format?

Comment (hellerve) [reply char.zp]: in this case: `- cmp: the comparison function` should be sufficient

File: char.zp (hunk 1 above)
Comment (hellerve) [reply char.zp:2]: So we can tell where the meta information starts (it is assumed to be the last paragraph).

**Comment by hellerve** [comment]
Perfect. I don't think that the parameter descriptions are all that helpful but to be honest, I wouldn't know how to better describe them.

**Pull request merged**